packages/kbn-transpose-helpers/index.ts

```diff
@@ -86,6 +94,11 @@ export function transposeTable(
         transposedColumnId,
         metricsColumnArgs
       );
+
+      const colOrderMap = new Map(args.columns.map((c, i) => [c.columnId, i]));
+      firstTable.columns.sort((a, b) => {
+        return (colOrderMap.get(a.id) ?? 0) - (colOrderMap.get(b.id) ?? 0);
+      });
```

Comment:
I believe this fixes #164413 as an alternate to sorting columns in csv exporter.